### ISSUE TITLE
[@typescript-eslint/quotes] False Positive in interface

### ISSUE THREAD

**Issue by RyanHirsch** [typescript-eslint/typescript-eslint#1309]
[@typescript-eslint/quotes] False positive in Typescript Interface

**Repro**

```JSON
{
  "rules": {
    "quotes": "off",
    "@typescript-eslint/quotes": ["error", "backtick"],
  }
}
```

```TS
export interface CardAttributes {
  id: number;
  name: string;
  creator: string;
  description: string;
  'is-joined': boolean;
  'is-joining': boolean;
  'is-approval-required': boolean;
}
```

**Expected Result**
No Errors

**Actual Result**
   6:3  error  Strings must use backtick  @typescript-eslint/quotes
   7:3  error  Strings must use backtick  @typescript-eslint/quotes
   8:3  error  Strings must use backtick  @typescript-eslint/quotes

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.10.0` |
| `@typescript-eslint/parser`        | `2.10.0` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.1.0` |
| `node`                             | `10.16.3` |
